Assets/Scripts/GridGen.cs

```diff
@@ -13,6 +14,7 @@ public class GridGen : MonoBehaviour
 
     [HideInInspector]
     public List<List<char>> tiles;
+    private Stack<List<List<char>>> states;
```

Comment:
maybe `history` or `stateHistory` would be a better name?
Since `state` is a bit generic enough it could mean different things